part3.md

```diff
@@ -0,0 +1,80 @@
+## Part 3 - Hello Kotlin
+
+ * Make sure that you have Kotlin JS installed (in `functions` sub-folder)
+```
+npm install -g kotlin
+```
+
+ * Open IntelliJ IDEA and create new pure Kotlin project in root folder. Use Kotlin(JavaScript) and Gradle in wizard. As a result you shold have `functions` folder as a subfolder of you project. 
+
+ * Add following block or `build.gradle`:
+``` groovy
+compileKotlin2Js.kotlinOptions {
+    moduleKind = "commonjs"
+    outputFile = "functions/build/index.js"
+}
+```
+
+ * Write same `testPush` function in `src/main/kotlin/Index.kt` in basic KotlinJS:
+``` kotlin
+external fun require(module: String): dynamic
+external val exports: dynamic
+
+val functions = require("firebase-functions")
+val admin = require("firebase-admin")
+
+data class Conversions(
+    val eur: Float,
+    val usd: Float? = null,
+    val czk: Float? = null,
+    val pln: Float? = null,
+    val rub: Float? = null,
+    val timestamp: Long? = null
+)
+
+fun main(args: Array<String>) {
+    admin.initializeApp(functions.config().firebase)
+
+    exports.testPush = functions.https.onRequest { req, res ->
+        admin.database().ref("/conversions")
+            .push(Conversions(req.query.eur.toString().toFloat()))
+            .then { res.status(200).send("done") }
+    }
+}
+```
+
+ * Try to build and deploy you function:
+``` sh
+./gradlew build
+firebase deploy --only function
+```
+
+It is possible that you will get `Error: Error parsing triggers: Cannot find module` error, in such case run following commands to reset dependencies:
```

Comment:
that you will get `message` error -> to encounter the error `message`/that you get the error `message`
, in such case -> . If that is the case, 
the/your dependencies